crates/wasmtime/src/trap.rs

```diff
@@ -183,7 +183,54 @@ pub struct WasmBacktrace {
 }
 
 impl WasmBacktrace {
-    fn new(
+    /// Captures a trace of the WebAssembly frames on the stack for the
+    /// provided store.
+    ///
+    /// This will return a [`WasmBacktrace`] which holds captured
+    /// [`FrameInfo`]s for each frame of WebAssembly on the call stack of the
+    /// current thread. If no WebAssembly is on the stack then the returned
+    /// backtrace will have no frames in it.
+    ///
+    /// Note that this function does not respect the [`Config::wasm_backtrace`]
+    /// configuration option and instead always captures a backtrace.
+    ///
+    /// [`Config::wasm_backtrace`]: crate::Config::wasm_backtrace
```

Comment:
Having two constructors, one forcing and one respecting the config, might be nice.